mobile/src/reducers/organization/employees.reducer.ts

```diff
@@ -19,30 +19,36 @@ const defaultState: EmployeesStore = {
 
 export const employeesReducer: Reducer<EmployeesStore> = (state = defaultState, action: OrganizationActions) => {
     switch (action.type) {
-        case 'LOAD_EMPLOYEE_FINISHED':
-            let {employeesById, employeeIdsByDepartment} = state;
-
-            const newEmployee = action.employee;
-            const oldEmployee = employeesById.get(newEmployee.employeeId);
-            if (oldEmployee !== undefined) {
-                //if department changed, remove a link from old collection
-                if (oldEmployee.departmentId !== newEmployee.departmentId) {
-                    employeeIdsByDepartment = employeeIdsByDepartment
-                        .update(oldEmployee.departmentId, Set(), oldColelction => oldColelction.remove(oldEmployee.employeeId));
+        case 'LOAD_EMPLOYEES_FINISHED':
+            let { employeesById, employeeIdsByDepartment } = state;
+
+            for (const newEmployee of action.employees) {
```

Comment:
action.employees.foreach(newEmployee=>{... seems to be better looking for me

Reply:
no, try to avoid functions. what's the point in creating a new function that hits performance, readability and debugging.